transformer_engine/common/include/transformer_engine/cast.h

```diff
@@ -270,6 +270,20 @@ void nvte_multi_tensor_quantize(const NVTETensor *inputs, NVTETensor *outputs,
                                 const NVTEQuantizationConfig quant_config, const size_t num_tensors,
                                 cudaStream_t stream);
 
+/*! \brief Casts grouped input tensor to quantized output tensors.
+ *
+ *  \param[in]      input           Input tensor to be cast.
+ *  \param[in,out]  outputs          Output quantized tensors.
+ *  \param[in]      split_sections   Split sections of the input tensor.
+ *  \param[in]      num_tensors      Number of output tensors.
+ *  \param[in]      quant_config    (Optional) Quantization configurations.
+ *  \param[in]      stream           CUDA stream used for the operation.
+ */
+void nvte_group_nvfp4_quantize_with_amax(const NVTETensor input, NVTETensor *outputs,
+                                         const size_t *split_sections, size_t num_tensors,
+                                         const NVTEQuantizationConfig quant_config,
+                                         cudaStream_t stream);
```

Comment:
- This API should be general and not specific to NVFP4.
- We should use the `GroupedTensor` API instead of CPU split points.